src/unstract/sdk/x2txt.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABCMeta
+
+from unstract.adapters.constants import Common
+from unstract.adapters.x2text import adapters
+from unstract.adapters.x2text.constants import X2TextConstants
+from unstract.adapters.x2text.x2text_adapter import X2TextAdapter
+
+from unstract.sdk.adapters import ToolAdapter
+from unstract.sdk.constants import LogLevel
+from unstract.sdk.tool.base import BaseTool
+
+
+class X2Text(metaclass=ABCMeta):
+    def __init__(self, tool: BaseTool):
+        self.tool = tool
+        self.x2text_adapters = adapters
+
+    def get_x2text(self, adapter_instance_id: str) -> X2TextAdapter:
+        try:
+            x2text_config = ToolAdapter.get_adapter_config(
+                self.tool, adapter_instance_id
+            )
+            x2text_adapter_id = x2text_config.get(Common.ADAPTER_ID)
+            if x2text_adapter_id in self.x2text_adapters:
+                x2text_adapter = self.x2text_adapters[x2text_adapter_id][
+                    Common.METADATA
+                ][Common.ADAPTER]
+                x2text_metadata = x2text_config.get(Common.ADAPTER_METADATA)
+                # Add x2text service host, port and platform_service_key
+                x2text_metadata[
+                    X2TextConstants.X2TEXT_HOST
+                ] = self.tool.get_env_or_die(X2TextConstants.X2TEXT_HOST)
+                x2text_metadata[
+                    X2TextConstants.X2TEXT_PORT
+                ] = self.tool.get_env_or_die(X2TextConstants.X2TEXT_PORT)
+                x2text_metadata[
+                    X2TextConstants.PLATFORM_SERVICE_API_KEY
+                ] = self.tool.get_env_or_die(
```

Comment:
@gaya3-zipstack will this work fine for PromptStudio in a multitenant env as well since it runs from the backend?

Reply:
@chandrasekharan-zipstack  If we are able to provide the platform service API key from the environment, then I don't see a problem.
PS: This is the same approach that is being followed for getting adapter config also from platform service.

Reply:
Ah now I remember - we follow [a hacky approach](https://github.com/Zipstack/Pandora/blob/b0b43085c9b881ab859691ccb12384d19412ffd8/backend/prompt_studio/prompt_studio_core/prompt_ide_base_tool.py#L33-L36) where we override this function and explicitly return the key.
I don't recommend this approach however it was a solution implemented for the sake of time